numpy/core/src/multiarray/dtype_transfer.c

```diff
@@ -3482,8 +3521,13 @@ PyArray_GetDTypeCopySwapFn(int aligned,
                                     itemsize);
         *outtransferdata = NULL;
     }
+    else if (dtype->kind == 'U') {
+        return wrap_copy_swap_function(aligned,
+                                       src_stride, dst_stride, dtype, 1,
+                                       outstransfer, outtransferdata);
```

Comment:
Sure, can change, not sure it should be faster, since I would expect this uses the copyswapn function and not copyswap which might actually be quicker, but did not check.